### PR TITLE
Disable filer UI in configuration

### PR DESCRIPTION
**Pull request by jerebear12**
# What problem are we solving?

There is currently no way to disable the Filer returning directory metadata when the Filer is exposed to the public. The previous best option was to require the JWT for reads and set the UI to readonly mode(which isn't always possible depending on the use case). This is good, but still exposes the paths and names of files. If the Filer is not set to readonly mode, then a client can request file paths via the Filer, view them, and a valid JWT, delete them. 

# How are we solving the problem?

This pull adds a configuration property that allows you to hide the Filer UI by setting the filer.expose_directory_metadata.enabled property to `false`. This will make it harder for malicious users to access the files in SeaweedFS. This way the only way the paths to images will be exposed is if your application exposes their routes. 

# How is the PR tested?

This was tested on my local machine with the configuration property set to `true` and `false`. 

# Checks
- [ ] I have added unit tests if possible.
- [x] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

File: weed/command/filer.go [hunk 1]
```diff
@@ -93,6 +94,7 @@ func init() {
 	f.downloadMaxMBps = cmdFiler.Flag.Int("downloadMaxMBps", 0, "download max speed for each download request, in MB per second")
 	f.diskType = cmdFiler.Flag.String("disk", "", "[hdd|ssd|<tag>] hard drive or solid state drive or any tag")
 	f.allowedOrigins = cmdFiler.Flag.String("allowedOrigins", "*", "comma separated list of allowed origins")
+	f.exposeDirectoryData = cmdFiler.Flag.Bool("exposeDirectoryData", true, "whether to return directory metadata and content in Filer UI")
```
Comment (chrislusf) [inline weed/command/filer.go:97]: usually server.go file also needs to add a corresponding option.

File: weed/server/filer_server.go [hunk 2]
```diff
@@ -115,6 +116,10 @@ func NewFilerServer(defaultMux, readonlyMux *http.ServeMux, option *FilerOption)
 	domains := strings.Split(allowedOrigins, ",")
 	option.AllowedOrigins = domains
 
+	v.SetDefault("filer.expose_directory_metadata.enabled", true)
```
Comment (jerebear12) [inline weed/server/filer_server.go:119]: @chrislusf Is this the option you are referring to in your comment above?

File: weed/command/filer.go (hunk 1 above)
Comment (jerebear12) [reply weed/command/filer.go:97]: See my comment below. I believe the server.go file referenced is the filer_server.go file?

File: weed/server/filer_server.go (hunk 2 above)
Comment (chrislusf) [reply weed/server/filer_server.go:119]: this file https://github.com/seaweedfs/seaweedfs/blob/e88467bb36b3618723c663d6b0e96541f654371c/weed/command/server.go#L109

Comment (jerebear12) [reply weed/server/filer_server.go:119]: Thank you, updated it.

**Comment by hencoappel** [comment]
@jerebear12 What is the intention here? I see exposeDirectoryData option has been added here but it's never passed to NewFilerServer() in filer.go?
Even if it was, it looks like it's always overwritten by the value from 'filer.expose_directory_metadata.enabled'?
Is this expected?

**Pull request merged**